File: geometric-classes-js/public/js/Circle.js
class Circle {
  constructor(radius, x, y) {
    this.radius = radius;
    this.x = x || 0;
    this.y = y || 0;
   }

diameter() {
 let dia = this.radius*2;
 return dia;
 }
area(){
  let area = Math.PI * (this.radius*this.radius);
  return area;
}
perimeter(){
  let peri = Math.PI * 2 * this.radius;
  return peri;
}
}



// def center_point
//   [@x, @y]
// end
//
// def area
//   Math::PI * @radius * @radius
// end
//
// def perimeter
//   Math::PI * 2.0 * @radius
// end
// def diameter
//   @radius * 2
//
// }

File: geometric-classes/lib/circle.rb
class Circle
  attr_reader :radius, :x, :y

  def initialize(radius = 1, x = 0, y = 0)
    @x = x
    @y = y
    @radius = radius
  end

  def center_point
    [@x, @y]
  end

  def area
    Math::PI * @radius * @radius
  end

  def perimeter
    Math::PI * 2.0 * @radius
  end
  def diameter
    @radius * 2
  end
end
circle = Circle.new(3)
require 'pry'
binding.pry
